dvc/repo/add.py

```diff
@@ -160,27 +160,33 @@ def _create_stages(repo, targets, fname, pbar=None, external=False):
         disable=len(targets) < LARGE_DIR_SIZE,
         unit="file",
     ):
-        path, wdir, out = resolve_paths(repo, out)
-        stage = create_stage(
-            Stage,
-            repo,
-            fname or path,
-            wdir=wdir,
-            outs=[out],
-            external=external,
-        )
-        if stage:
-            Dvcfile(repo, stage.path).remove()
+        import glob
+
+        outs = glob.glob(out)
```

Comment:
Expanded out means more targets, so looks like we should do that before starting the `targets` loop/progress-bar above.